dotnet/src/SemanticKernel/Memory/Collections/TopNSortedList.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.SemanticKernel.Memory.Collections;
+
+/// <summary>
+/// A sorted list that only keeps the top N items.
+/// </summary>
+/// <typeparam name="T">Type of the item.</typeparam>
+internal class TopNSortedList<T> : SortedList<double, IEnumerable<T>>
+{
+    /// <summary>
+    /// Creates an instance of the <see cref="TopNSortedList{T}"/> class.
+    /// </summary>
+    /// <param name="maxSize">The maximum number of items this collection can store.</param>
+    public TopNSortedList(int maxSize)
+        : base(new DescendingDoubleComparer())
+    {
+        this._maxSize = maxSize;
+    }
+
+    /// <summary>
+    /// Adds a new item to the list.
+    /// </summary>
+    /// <param name="score">The item's score.</param>
+    /// <param name="value">The item's value.</param>
+    public void Add(double score, T value)
+    {
+        if (this.Count >= this._maxSize)
```

Comment:
this.Count is the number of collections of items, right? Not the total number of items? So the maxSize is not strictly followed because with duplicates, it'll be (n x m) items in the whole collection.

Reply:
Yes, that's correct, good catch! I will add `TopNCollection` and `MinHeap` classes instead of this one.